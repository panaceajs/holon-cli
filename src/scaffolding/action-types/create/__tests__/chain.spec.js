const { resolve } = require('path');
const snapshot = require('../../../../utils/task-snapshot');
const task = require('../chain');

describe('action-types', () => {
  it('should create action types with tests in `action-types`', async () => {
    await expect(
      snapshot(task, {
        stateNamespace: 'test',
        actionTypes: { anAction: 'anAction', anotherAction: 'anotherAction' }
      })
    ).resolves.toMatchSnapshot();
  });
  it('should create action types and actions with tests', async () => {
    await expect(
      snapshot(task, {
        magic: true,
        stateNamespace: 'test',
        actionTypes: { anAction: 'anAction', anotherAction: 'anotherAction' }
      })
    ).resolves.toMatchSnapshot();
  });
  it('should create action types and actions with tests in `my-dir`', async () => {
    await expect(
      snapshot(task, {
        cwd: resolve('other-dir'),
        magic: true,
        stateNamespace: 'test',
        actionTypes: { anAction: 'anAction', anotherAction: 'anotherAction' }
      })
    ).resolves.toMatchSnapshot();
  });
});
