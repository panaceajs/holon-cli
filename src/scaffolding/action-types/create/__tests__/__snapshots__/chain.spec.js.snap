// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`action-types should create action types and actions with tests 1`] = `
"
--------------------------------------------------------------------------------
// ./action-types/__tests__/index.spec.js
// Create
--------------------------------------------------------------------------------

import { AN_ACTION, ANOTHER_ACTION } from '../';

describe('\`test\` action types', () => {
  it('should export \`AN_ACTION\` type', () => {
    expect(AN_ACTION).toBe('AN_ACTION');
  });

  it('should export \`ANOTHER_ACTION\` type', () => {
    expect(ANOTHER_ACTION).toBe('ANOTHER_ACTION');
  });
});







--------------------------------------------------------------------------------
// ./action-types/index.js
// Create
--------------------------------------------------------------------------------

export const AN_ACTION = 'AN_ACTION';
export const ANOTHER_ACTION = 'ANOTHER_ACTION';







--------------------------------------------------------------------------------
// ./actions/__tests__/index.spec.js
// Create
--------------------------------------------------------------------------------

import { AN_ACTION, ANOTHER_ACTION } from '../../action-types';
import { anAction, anotherAction } from '../';

describe('\`test\` action creators', () => {
  it('should export \`anAction\` function', () => {
    expect(anAction).toBeDefined();
    expect(anAction()).toEqual({
      type: AN_ACTION
    });
  });

  it('should export \`anotherAction\` function', () => {
    expect(anotherAction).toBeDefined();
    expect(anotherAction()).toEqual({
      type: ANOTHER_ACTION
    });
  });
});







--------------------------------------------------------------------------------
// ./actions/__tests__/index.spec.js
// Create
--------------------------------------------------------------------------------

import { AN_ACTION, ANOTHER_ACTION } from '../../action-types';
import { anAction, anotherAction } from '../';

describe('\`test\` action creators', () => {
  it('should export \`anAction\` function', () => {
    expect(anAction).toBeDefined();
    expect(anAction()).toEqual({
      type: AN_ACTION
    });
  });

  it('should export \`anotherAction\` function', () => {
    expect(anotherAction).toBeDefined();
    expect(anotherAction()).toEqual({
      type: ANOTHER_ACTION
    });
  });
});





"
`;

exports[`action-types should create action types and actions with tests in \`my-dir\` 1`] = `
"
--------------------------------------------------------------------------------
// ./action-types/__tests__/index.spec.js
// Create
--------------------------------------------------------------------------------

import { AN_ACTION, ANOTHER_ACTION } from '../';

describe('\`test\` action types', () => {
  it('should export \`AN_ACTION\` type', () => {
    expect(AN_ACTION).toBe('AN_ACTION');
  });

  it('should export \`ANOTHER_ACTION\` type', () => {
    expect(ANOTHER_ACTION).toBe('ANOTHER_ACTION');
  });
});







--------------------------------------------------------------------------------
// ./action-types/index.js
// Create
--------------------------------------------------------------------------------

export const AN_ACTION = 'AN_ACTION';
export const ANOTHER_ACTION = 'ANOTHER_ACTION';







--------------------------------------------------------------------------------
// ./actions/__tests__/index.spec.js
// Create
--------------------------------------------------------------------------------

import { AN_ACTION, ANOTHER_ACTION } from '../../action-types';
import { anAction, anotherAction } from '../';

describe('\`test\` action creators', () => {
  it('should export \`anAction\` function', () => {
    expect(anAction).toBeDefined();
    expect(anAction()).toEqual({
      type: AN_ACTION
    });
  });

  it('should export \`anotherAction\` function', () => {
    expect(anotherAction).toBeDefined();
    expect(anotherAction()).toEqual({
      type: ANOTHER_ACTION
    });
  });
});







--------------------------------------------------------------------------------
// ./actions/__tests__/index.spec.js
// Create
--------------------------------------------------------------------------------

import { AN_ACTION, ANOTHER_ACTION } from '../../action-types';
import { anAction, anotherAction } from '../';

describe('\`test\` action creators', () => {
  it('should export \`anAction\` function', () => {
    expect(anAction).toBeDefined();
    expect(anAction()).toEqual({
      type: AN_ACTION
    });
  });

  it('should export \`anotherAction\` function', () => {
    expect(anotherAction).toBeDefined();
    expect(anotherAction()).toEqual({
      type: ANOTHER_ACTION
    });
  });
});





"
`;

exports[`action-types should create action types with tests in \`action-types\` 1`] = `
"
--------------------------------------------------------------------------------
// ./action-types/__tests__/index.spec.js
// Create
--------------------------------------------------------------------------------

import { AN_ACTION, ANOTHER_ACTION } from '../';

describe('\`test\` action types', () => {
  it('should export \`AN_ACTION\` type', () => {
    expect(AN_ACTION).toBe('AN_ACTION');
  });

  it('should export \`ANOTHER_ACTION\` type', () => {
    expect(ANOTHER_ACTION).toBe('ANOTHER_ACTION');
  });
});







--------------------------------------------------------------------------------
// ./action-types/index.js
// Create
--------------------------------------------------------------------------------

export const AN_ACTION = 'AN_ACTION';
export const ANOTHER_ACTION = 'ANOTHER_ACTION';





"
`;
